diff --git a/core/modules/content_moderation/src/Plugin/views/field/ModerationStateField.php b/core/modules/content_moderation/src/Plugin/views/field/ModerationStateField.php
index 668d40a708..869ecbe99e 100644
--- a/core/modules/content_moderation/src/Plugin/views/field/ModerationStateField.php
+++ b/core/modules/content_moderation/src/Plugin/views/field/ModerationStateField.php
@@ -30,6 +30,10 @@ public function clickSort($order) {
     $column_name = $storage->getTableMapping()->getFieldColumnName($storage_definition, 'value');
     if (!isset($this->aliases[$column_name])) {
       $this->aliases[$column_name] = $this->tableAlias . '.' . $column_name;
+      // If the query uses DISTINCT we need to add the column too.
+      if (!empty($this->view->getQuery()->options['distinct'])) {
+        $this->query->addField($this->tableAlias, $column_name);
+      }
       $this->query->addOrderBy(NULL, NULL, $order, $this->aliases[$column_name]);
     }
   }
